compiler/rustc_codegen_llvm/src/metadata.rs

```diff
@@ -18,51 +19,66 @@ pub use rustc_data_structures::sync::MetadataRef;
 pub struct LlvmMetadataLoader;
 
 impl MetadataLoader for LlvmMetadataLoader {
-    fn get_rlib_metadata(&self, _: &Target, filename: &Path) -> Result<MetadataRef, String> {
+    fn get_rlib_metadata(&self, target: &Target, filename: &Path) -> Result<MetadataRef, String> {
```

Comment:
FWIW I think it may be high time to rip out all the LLVM integration here and simply use the `object` crate (or similar) to read the object files from the archives here (maybe even use the `ar` crate!). I'm not certain about this though so I figured it was best left to a different PR. For now this continues to use all the LLVM integration we're already used to.

Put another way, I think it might be good for a future PR to explore simply lifting the probably-existing cranelift solution for this into the complier and remove the need for this to be a backend-specific trait.

Reply:
I tried to use the `object` crate for both object file and archive reading in https://github.com/rust-lang/rust/pull/83640, but it regressed performance a bit.

Reply:
I rebased and reopened #83640.